src/main/java/bio/terra/service/tabulardata/google/bigquery/BigQueryDatasetPdao.java

```diff
@@ -565,6 +629,64 @@ public void insertIntoDatasetTable(
                 Optional.ofNullable(transactId).map(UUID::toString).orElse(null))));
   }
 
+  private static final String insertNewFileIdsIntoDatasetTableTemplate =
+      """
+    /* Soft delete all existing rows */
+    INSERT INTO `<project>.<dataset>.<softDeleteTable>` (<transactIdColumn>, <rowIdColumn>, <flightIdColumn>, <deletedAtColumn>, <deletedByColumn>)
+    SELECT @transactId, <rowIdColumn>, @flightId, CURRENT_TIMESTAMP(), @deletedBy
+    FROM `<project>.<dataset>.<liveTable>`;
+
+    /* Insert rows with new values */
+    INSERT INTO `<project>.<dataset>.<targetTable>` (<transactIdColumn>, <rowIdColumn>, <columns:{c|<c.name>}; separator=",">)
+    SELECT @transactId,GENERATE_UUID(),<columns:{c|<if(c.fileOrDirRef)>
+      <if(c.arrayOf)>
+        ARRAY(SELECT f.<newIdColumn> FROM UNNEST(<c.name>) i JOIN `<project>.<dataset>.<fileIdMappingTable>` f ON i = f.<origIdColumn>)
+      <else>
+        (SELECT <newIdColumn> FROM `<project>.<dataset>.<fileIdMappingTable>` f WHERE f.<origIdColumn> = t.<c.name>)
+      <endif>
+    <else>
+      <c.name>
+    <endif>}; separator=",">
+    FROM `<project>.<dataset>.<liveTable>` t;
+      """;
+
+  public void insertNewFileIdsIntoDatasetTable(
+      Dataset dataset,
+      DatasetTable targetTable,
+      UUID transactId,
+      String flightId,
+      AuthenticatedUserRequest authedUser)
+      throws InterruptedException {
+    BigQueryProject bigQueryProject = BigQueryProject.from(dataset);
+
+    ST sqlTemplate = new ST(insertNewFileIdsIntoDatasetTableTemplate);
```

Comment:
Personal preference but I've been finding it more readable to chain the ST parameter specs, like so:
```
ST sqlTemplate = new ST(insertNewFileIdsIntoDatasetTableTemplate)
  .add("project", bigQueryProject.getProjectId())
  .add("dataset", BigQueryPdao.prefixName(dataset.getName()))
 ….
```

Reply:
ah, that's a great call.  I'll change to the fluent style

Reply:
I'll actually do it for all of the `BigQuery*Pdao` classes (multiline editing FTW!)